docs/general/administration/hardware-acceleration/apple.md

```diff
@@ -0,0 +1,102 @@
+
+Full acceleration is available on most Macs from 2017 and later, with the exception of the MacBook Air (13-inch, 2017).
+
+To have native Apple Silicon support, Jellyfin server 10.9.0+ and `jellyfin-ffmpeg6` 6.0.1-5 or higher is required.
+
+While hardware acceleration via VideoToolbox might work on older series Macs, it is not officially supported.
+
+<sup>*</sup> VideoToolbox may fallback to software decoding/encoding or stop working when the task exceeds its hardware capability. The older your Mac, the more this will occur. Macs from 2017 and later, especially Apple Silicon Macs are strongly recommended.
+
+### Transcode H.264
+
+AVC / H.264 8-bit is still widely used due to its excellent compatibility. Most GPUs after 2011 can decode and encode it.
+
+Any VideoToolbox-supported Mac supports decoding and encoding H.264 8-bit.
+
+### Transcode HEVC
+
+HEVC / H.265 remains the first choice for storing 4K 10-bit, HDR and Dolby Vision video. It has mature software encoding support thanks to [x265](https://x265.readthedocs.io/en/master/), as well as the widely implemented hardware encoding support in most GPUs released after 2016.
+
+Macs from 2017 and later, excluding the MacBook Air (13-inch, 2017), support decoding and encoding HEVC.
+
+### Transcode AV1
+
+There is no hardware accelerated path for AV1 on macOS at the moment. Although the M3 series added AV1 decoding support, [ffmpeg does not support it yet](https://trac.ffmpeg.org/ticket/10642).
+
+### Performance Consideration
+
+An Apple Silicon-based Mac is preferred in most cases. Even an entry-level M1 can handle three 4K 24fps Dolby Vision HEVC 10-bit transcoding tasks simultaneously while performing tone-mapping to SDR.<sup>1</sup> <sup>2</sup>
+
+The "Max" variant chips come with an additional video encoding engine. VideoToolbox can utilize this extra engine even when there is only a single transcoding session, enabling support for 4K 120fps transcoding and tone-mapping.<sup>1</sup>
+
+The "Ultra" variant chips feature 2 video decoding engines and 4 video encoding engines, effectively doubling the capability compared to the "Max" variant chips.
+
+On legacy Intel Macs, you may encounter performance issues with tone-mapping using Metal on 4K videos if your Mac doesn't have an AMD GPU. However, it is still adequate for transcoding SDR videos.
+
+<sup>1</sup> Using prefer speed encoder preset.
+
+<sup>2</sup> The simultaneous session count is a soft limit. You can run more sessions if you want, but the transcoding performance of each session may be reduced to a point where video playback starts to stutter.
+
+## macOS Setups
+
+**macOS 12 and later** are officially supported. Older versions might work, but are not supported.
+
+### Configure
+
+- Enable VideoToolbox in Jellyfin and uncheck unsupported codecs.
+  - Disable `MPEG4`, `MPEG2`, and `VC-1`, as VideoToolbox lacks a hardware-accelerated decoding path for these codecs.
```

Comment:
TBH you can eliminate these legacy hw codecs in the server and web. I don't really believe Apple will introduce them in the future.

Reply:
We also need to perform some server-side hacks to revert those options if the user enabled them before.

These options have been available for a long time and for a reason. Previously, ffmpeg allowed software fallback for all codecs, and these legacy codecs would use VideoToolbox's software fallback route. However, now ffmpeg only allows fallback for HEVC, and these codecs will error out because VT session creation would fail.

Is such change still good for 10.9?

Reply:
We need a codec panel that can be checked automatically. At that time, the existing decoder/encoder settings will be discarded and Jellyfin will be automatically set based on the hardware probing results.

So deleting these codecs should have no impact.
https://github.com/jellyfin/jellyfin/blob/1124da5428b4cb0e26133f636b27634723026513/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs#L6167-L6175

Reply:
Updated doc and pushed changes to server: jellyfin/jellyfin#11217 and web: jellyfin/jellyfin-web#5325